src/transformers/onnx/convert.py

```diff
@@ -62,90 +62,190 @@ def check_onnxruntime_requirements(minimum_version: Version):
-                model,
-                (model_inputs,),
-                f=output.as_posix(),
-                input_names=list(config.inputs.keys()),
-                output_names=onnx_outputs,
-                dynamic_axes={name: axes for name, axes in chain(config.inputs.items(), config.outputs.items())},
-                do_constant_folding=True,
-                opset_version=opset,
-            )
+    import tensorflow as tf
+
+    import onnx
+    import tf2onnx
+
+    model.config.return_dict = True
 
-        config.restore_ops()
+    # Check if we need to override certain configuration item
+    if config.values_override is not None:
+        logger.info(f"Overriding {len(config.values_override)} configuration item(s)")
+        for override_config_key, override_config_value in config.values_override.items():
+            logger.info(f"\t- {override_config_key} -> {override_config_value}")
+            setattr(model.config, override_config_key, override_config_value)
+
+    # Ensure inputs match
+    model_inputs = config.generate_dummy_inputs(tokenizer, framework=TensorType.TENSORFLOW)
+    inputs_match, matched_inputs = ensure_model_and_config_inputs_match(model, model_inputs.keys())
+    onnx_outputs = list(config.outputs.keys())
+
+    input_signature = [tf.TensorSpec.from_tensor(tensor, name=key) for key, tensor in model_inputs.items()]
+    onnx_model, _ = tf2onnx.convert.from_keras(model, input_signature, opset=opset)
+    onnx.save(onnx_model, output.as_posix())
+    config.restore_ops()
 
     return matched_inputs, onnx_outputs
 
 
+def export(
+    tokenizer: PreTrainedTokenizer,
+    model: Union[PreTrainedModel, TFPreTrainedModel],
+    config: OnnxConfig,
+    opset: int,
+    output: Path,
+) -> Tuple[List[str], List[str]]:
+    """
+    Export a Pytorch or TensorFlow model to an ONNX Intermediate Representation (IR)
+
+    Args:
+        tokenizer ([`PreTrainedTokenizer`]):
+            The tokenizer used for encoding the data.
+        model ([`PreTrainedModel`] or [`TFPreTrainedModel`]):
+            The model to export.
+        config ([`~onnx.config.OnnxConfig`]):
+            The ONNX configuration associated with the exported model.
+        opset (`int`):
+            The version of the ONNX operator set to use.
+        output (`Path`):
+            Directory to store the exported ONNX model.
+
+    Returns:
+        `Tuple[List[str], List[str]]`: A tuple with an ordered list of the model's inputs, and the named inputs from
+        the ONNX configuration.
+    """
+    if not (is_torch_available() or is_tf_available()):
+        raise ImportError(
+            "Cannot convert because neither PyTorch nor TensorFlow are not installed. "
+            "Please install torch or tensorflow first."
+        )
+
+    if is_torch_available():
+        return export_pytorch(tokenizer, model, config, opset, output)
+    elif is_tf_available():
+        return export_tensorflow(tokenizer, model, config, opset, output)
```

Comment:
The user might have both frameworks installed, and want to export a TF model. Should we check the type of the model instead (`PreTrainedModel`/`TFPreTrainedModel`)?

Reply:
Good idea! Maybe we can have something like:

```python
if issubclass(type(model), PreTrainedModel):
    return export_pytorch(tokenizer, model, config, opset, output)
elif issubclass(type(model), TFPreTrainedModel):
    return export_tensorflow(tokenizer, model, config, opset, output)
```

Reply:
I think the failing test is unrelated to the change - I've just re-run the job to see if it now passes